src/main/java/fi/aalto/cs/apluscourses/ui/module/ModuleListElementView.java

```diff
@@ -1,28 +1,49 @@
 package fi.aalto.cs.apluscourses.ui.module;
 
-import fi.aalto.cs.apluscourses.ui.GuiObject;
+import com.intellij.ui.ColoredListCellRenderer;
+import com.intellij.ui.SimpleTextAttributes;
+import com.intellij.ui.speedSearch.SpeedSearchUtil;
+import fi.aalto.cs.apluscourses.presentation.module.ModuleListElementViewModel;
 import fi.aalto.cs.apluscourses.ui.base.ListElementView;
 import icons.PluginIcons;
+import java.awt.Color;
 import javax.swing.JComponent;
-import javax.swing.JLabel;
-import javax.swing.JPanel;
+import javax.swing.JList;
+
+import org.jetbrains.annotations.NotNull;
 
 public class ModuleListElementView implements ListElementView {
-  @GuiObject
-  public JPanel basePanel;
-  @GuiObject
-  public JLabel nameLabel;
-  @GuiObject
-  public JLabel statusLabel;
-  @GuiObject
-  public JLabel updateLabel;
-
-  public ModuleListElementView() {
-    nameLabel.setIcon(PluginIcons.A_PLUS_MODULE);
+  private static class ColoredModuleListRenderer
+          extends ColoredListCellRenderer<ModuleListElementViewModel> {
+
+    private static final SimpleTextAttributes STATUS_TEXT_STYLE = new SimpleTextAttributes(
+            SimpleTextAttributes.STYLE_ITALIC | SimpleTextAttributes.STYLE_SMALLER, null);
+    private static final SimpleTextAttributes UPDATE_TEXT_STYLE = new SimpleTextAttributes(
+            SimpleTextAttributes.STYLE_BOLD, new Color(254, 127, 156));
```

Comment:
Does the color look good in both light and dark themes? [`JBColor`](https://upsource.jetbrains.com/idea-ce/file/idea-ce-4b94ba01122752d7576eb9d69638b6e89d1671b7/platform/util/ui/src/com/intellij/ui/JBColor.java) provides an alternative that contains two colors

Reply:
Will check, if it doesn't look good I'll replace it with the standard text color that's currently used

Reply:
The color certainly makes the update stand out, which is good, but it also almost gives the impression that the module has some kind of error
![image](https://user-images.githubusercontent.com/30376763/106513315-7cd6cf00-64db-11eb-9cb0-11aff5240861.png)

Reply:
To be fair I copied the color code from the current ModuleListElementView Swing form, but for some reason the color code is not shown in the actual UI

Reply:
Ah true, weird that it was never visible then :smile: